Fix infgrid artifacts for ortho camera

almarklein opened the pull request:
Fixes #821

Confirmed on Mac and Windows Intel.

hmaarrfk commented:
beautiful. it used to take me 0.1 seconds to break it, now my 10 seconds of scrolling are without any obvious shader artifacts!!!

Korijn reviewed (approved):
I wish I understood why this fixes the issue!

hmaarrfk commented on pygfx/renderers/wgpu/wgsl/grid.wgsl:
```diff
@@ -64,9 +66,21 @@ fn vs_main(in: VertexInput) -> Varyings {
+        // With an orthographic camera, we use the scale factors for x and y (camera width and height).
+        // For perspective cameras we use the distance to the plane and assume an aspect ratio of 1.
+        var near_multiplier = vec2<f32>(1.0);
+        if is_orthographic() {
```
to improve understanding, maybe this if statement should be around the whole computation for `distance_cam_to_grid`.

I understand that it won't help performance, but it may help the understanding in saying that:

"Our computation for the distance between the camera to the grid is only valid when the camera isn't an orthographic projection"

My understanding is that the nature of the orthographic projection makes the camera matrix illposed in certain situations.

almarklein commented:
I updated the comments to make it more clear, and moved the distance calcualtion into the if.

I found that the artifacts show up because of the big difference in scale between the x and y axis. The solution is basically to scale the grid in the same way. While writing the comments I realized that the solution does make the assumption that the grid is aligned with the camera. But I think this is a safe assumption for cases where you have a non-unit aspect ratio.
